File: my_first_cpp/src/my_first_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
using namespace std::chrono_literals;
class RobotNewsStationNode : public rclcpp::Node 
{
public:
    RobotNewsStationNode() : Node("robot_news_station"),sister_name_("neha") {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news_msg",10);
        timer_ = this->create_wall_timer(0.5s,std::bind(&RobotNewsStationNode::publishNews,this));
        RCLCPP_INFO(this->get_logger(),"Robot is publisheing:");
    }
private:
    void publishNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string(sister_name_+ std::string(" is my sis."));
        publisher_->publish(msg);
    }
    std::string sister_name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_first_py/my_first_py/my_first_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewsStationNode(Node):
    def __init__(self,sister_name):
        super().__init__("robot_news_station")
        self.publisher_ = self.create_publisher(String,"robot_news_msg",10)
        self.timer_ = self.create_timer(0.5,self.publisher_news)
        self.sister_name = sister_name
        self.get_logger().info("Robot News Station Is Publishing:")

    def publisher_news(self):
        msg= String()
        msg.data = f"{self.sister_name} Is My Sis."
        self.publisher_.publish(msg)

def main(args=None):
    sister_name = input("Entre Your Sister's Name:")
    rclpy.init(args=args)
    node = RobotNewsStationNode(sister_name)
    rclpy.spin(node)
    rclpy.shutdown(node)

if __name__== "__main__":
    main()

